pm_kext/src/pm_callouts.c

```diff
@@ -54,14 +54,38 @@ NTSTATUS initCalloutStructure() {
     return STATUS_SUCCESS;
 }
 
+// Used for freeing the packet info memory when clearing the packet cache
+static void freePacketInfo(PortmasterPacketInfo *info, verdict_t verdict) {
+    UNREFERENCED_PARAMETER(verdict);
+    if(info != NULL) {
+        portmasterFree(info);
+    }
+}
+
+static void freePacketInfoAndData(PortmasterPacketInfo *info, void *data) {
+    if(info != NULL && data != NULL) {
+        portmasterFree(info);
+        portmasterFree(data);
+    }
```

Comment:
Would it make sense to check for these separately?

Reply:
I don't think so. If one of them is not NULL and the other one is, then something is wrong.
And calling portmasterFree on invalid pointer will probably crash.